config/import-control.xml

```diff
@@ -146,13 +146,15 @@
     <allow pkg="com.puppycrawl.tools.checkstyle.utils"/>
   </subpackage>
 
-  <subpackage name="gui">
+  <subpackage name="gui" strategyOnMismatch="disallowed">
+    <allow pkg="antlr"/>
     <allow pkg="java.awt"/>
+    <allow pkg="java.io"/>
     <allow pkg="javax.swing"/>
-    <allow pkg="com.puppycrawl.tools.checkstyle.utils" local-only="true"/>
-    <allow class="com.puppycrawl.tools.checkstyle.TreeWalker" local-only="true"/>
+    <allow pkg="java.util" exact-match="true"/>
+    <allow pkg="com\.puppycrawl\.tools\.checkstyle\.(api|utils)" regex="true"/>
+    <allow class="com\.puppycrawl\.tools\.checkstyle\.(TreeWalker|JavadocDetailNodeParser)" regex="true"/>
```

Comment:
There is no need in regular expression. @romani is right. Lets keep 'allow' elements for each import.